### PR TITLE
feature/LP-4-Community-Wall

### PR DESCRIPTION
**Pull request by irv205**
CommunityWall View:

image

- Created the community Wall view, added a mock data list
- A Post model was created to emulate the data entry and be able to emulate the list.

### CODE REVIEW

File: feature/communitywall/src/main/java/com/iteneum/communitywall/domain/Post.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.iteneum.communitywall.domain
+
+//Mock Model
+data class Post(
```
Comment (melissaquezada94) [inline feature/communitywall/src/main/java/com/iteneum/communitywall/domain/Post.kt:19]: Could you please add documentation?

File: feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import com.iteneum.communitywall.domain.Post
+import com.iteneum.designsystem.components.LpPostCard
+
+@Composable
```
Comment (melissaquezada94) [inline feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt:30]: could you please add documentation?

File: feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        Post("https://img.favpng.com/25/7/23/computer-icons-user-profile-avatar-image-png-favpng-LFqDyLRhe3PBXM0sx2LufsGFU.jpg","Martin Perroni","10m","This is a post about something"),
+    )
+
+    Box(modifier = Modifier
+        .fillMaxSize()
+        .wrapContentSize(Alignment.Center)
+    ){
+        LazyColumn{
+            items(list){
+                LpPostCard(modifier = Modifier, it.name, it.img, it.time, it.message, {}, {})
```
Comment (melissaquezada94) [inline feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt:73]: Please add the name of each param for the function LpPostCard

File: feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        Post("https://img.favpng.com/25/7/23/computer-icons-user-profile-avatar-image-png-favpng-LFqDyLRhe3PBXM0sx2LufsGFU.jpg","Martin Perroni","10m","This is a post about something"),
+        Post("https://img.favpng.com/25/7/23/computer-icons-user-profile-avatar-image-png-favpng-LFqDyLRhe3PBXM0sx2LufsGFU.jpg","Martin Perroni","10m","This is a post about something"),
+    )
+
+    Box(modifier = Modifier
+        .fillMaxSize()
+        .wrapContentSize(Alignment.Center)
+    ){
+        LazyColumn{
+            items(list){
```
Comment (melissaquezada94) [inline feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt]: Please define a name for each item

File: feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt (hunk 3 above)
Comment (irv205) [reply feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt:73]: The parameters of the LSPostCard function are defined in the specific component.

File: feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt (hunk 4 above)
Comment (irv205) [reply feature/communitywall/src/main/java/com/iteneum/communitywall/presentation/CommunityWallView.kt]: The keyword **items** is an inline function imported from:
`import androidx.compose.foundation.lazy.items`
and that inline is passed the list in this case of POST

File: feature/communitywall/src/main/java/com/iteneum/communitywall/domain/Post.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+@property message The content of the message in the post.
+@author Irving Ulises Gonzalez Uscanga
+ **/
+
+data class Post(
+    val img : String,
+    val name : String,
+    val time : String,
+    val message : String
+)
```
Comment (carlosph816) [inline feature/communitywall/src/main/java/com/iteneum/communitywall/domain/Post.kt:24]: What do you think to use the this data class ItemList already exist in data and to add the new params? we could reuse it

Comment (irv205) [reply feature/communitywall/src/main/java/com/iteneum/communitywall/domain/Post.kt:24]: We can, but the model is a bit different, so I prefer to use my own model.

image

**Review by carlosph816 — APPROVED** [review]

**Review by melissaquezada94 — APPROVED** [review]

**Pull request merged**